ironfish-rust/src/note.rs

```diff
@@ -209,10 +209,13 @@ impl<'a> Note {
             &mut bytes_to_encrypt[SCALAR_SIZE..(SCALAR_SIZE + AMOUNT_VALUE_SIZE)],
         );
         bytes_to_encrypt[(SCALAR_SIZE + AMOUNT_VALUE_SIZE)..].copy_from_slice(&self.memo.0[..]);
-        let mut encrypted_bytes = [0; ENCRYPTED_NOTE_SIZE + aead::MAC_SIZE];
-        aead::encrypt(shared_secret, &bytes_to_encrypt, &mut encrypted_bytes);
-
-        encrypted_bytes
+        let encrypted_bytes = aead::encrypt(shared_secret, &bytes_to_encrypt)?;
+        assert_eq!(encrypted_bytes.len(), ENCRYPTED_NOTE_SIZE + aead::MAC_SIZE);
+        
+        Ok(encrypted_bytes
+            .try_into()
```

Comment:
What type should this error be? Should I create a type? The type is going from `Vec<u8>` to `&[u8; 32]`

Reply:
I think it's fine being generous with error types, maybe like `IronfishError::InvalidEncryptedNoteLength`. I don't think our error types are especially well thought-out so far, so I'm not too concerned with getting it right here

Reply:
Wow that is cool syntax @mat-if, thanks for the tip!